src/java.base/share/classes/java/net/URL.java

```diff
@@ -352,7 +361,9 @@ public final class URL implements java.io.Serializable {
      * @param      port       the port number on the host.
      * @param      file       the file on the host
      * @throws     MalformedURLException  if an unknown protocol or the port
-     *                  is a negative number other than -1
+     *                  is a negative number other than -1, or if the
+     *                  underlying stream handler implementation rejects,
+     *                  or is known to reject, the {@code URL}
```

Comment:
and "underlying stream handling implementation rejects" would fit with the constructor specifying that it calls the stream handler's parseURL method.

Reply:
Done